### PR TITLE
LPD-23206 [Marketplace] Create Solution Publishing Context

### PR DESCRIPTION
**Pull request by joanagrego**

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Profile.tsx [hunk 1]
```diff
@@ -3,117 +3,253 @@
...
+						required
+					>
+						{i18n.translate('tags')}
+					</Form.Label>
+
+					<ClayMultiSelect
+						{...{placeholder: 'Select tags'}}
+						inputName="tags-selector"
+						items={tags}
+						key={'tags-' + tags.length}
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Profile.tsx:230]: Could you use template string on this
```suggestion
						items={tags}
						key={`tags-${tags.length}`}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Profile.tsx [hunk 2]
```diff
@@ -3,117 +3,253 @@
...
-					sourceItems={sourceItems}
-					spritemap={getIconSpriteMap()}
-					value={value}
-				/>
-			</div>
+					<ClayMultiSelect
+						{...{placeholder: 'Select categories'}}
+						inputName="description-selector"
+						items={categories}
+						key={'cat-' + categories.length}
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Profile.tsx:195]: Could you use template string on this
```suggestion
						items={categories}
						key={`cat-${categories.length}`}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Profile.tsx [hunk 3]
```diff
@@ -3,117 +3,253 @@
...
-		label: 'Category',
-		value: 'Category',
-	},
-	{
-		label: 'Tag',
-		value: 'Tag',
-	},
-];
+const tooltipInfo = {
+	categories:
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Solutions/NewSolutionFlow/pages/Profile.tsx:24]: Could you create the translations for this

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/hooks/data/useGetVocabulariesAndCategories.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	return useSWR({key: 'vocabularies/', vocabulariesName}, async () => {
+		const {
+			items,
+		} = await HeadlessCommerceAdminCatalogImpl.getMarketplaceVocabularies();
+
+		const _vocabularies = items.filter((vocabulary: ProductCategories) =>
+			vocabulariesName.includes(vocabulary.name as ProductVocabulary)
+		);
+
+		const vocabularies: any = {};
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/hooks/data/useGetVocabulariesAndCategories.ts:21]: Could you create the type for this

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+		description: '';
+		file: {};
+		name: '';
+		tags: [];
+	}>;
+};
+
+export type AppActions = ActionMap<SolutionPayload>[keyof ActionMap<
+	SolutionPayload
+>];
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/SolutionContext.tsx:54]: Break the line

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/UploadLogo/UploadLogo.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+						Upload Image
+					</label>
+				</div>
+
+				{uploadedFile?.uploaded && (
+					<button
+						className="btn btn-secondary btn-sm m-0 upload-logo-delete-button"
+						onClick={() => onDeleteFile(uploadedFile.id)}
+					>
+						Delete
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/UploadLogo/UploadLogo.tsx:73]: Could you use i18n

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/UploadLogo/UploadLogo.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+							}
+						}}
+						type="file"
+					/>
+
+					<label
+						className="btn btn-primary btn-sm m-0"
+						htmlFor="file"
+					>
+						Upload Image
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/UploadLogo/UploadLogo.tsx:64]: Could you use i18n

**Review by welitonpe — CHANGES_REQUESTED** [review]

**Review by kevenleone — CHANGES_REQUESTED** [review]
@joanagrego take a look please

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/hooks/data/useGetVocabulariesAndCategories.ts (hunk 4 above)
Comment (joanagrego) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/hooks/data/useGetVocabulariesAndCategories.ts:21]: We're gonna fix it in the next PR. Thanks!

**Comment by joanagrego** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/services/rest/HeadlessCommerceAdminTaxonomy.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Liferay} from '../../liferay/liferay';
+import fetcher from '../fetcher';
+
+class HeadlessCommerceAdminTaxonomy {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/services/rest/HeadlessCommerceAdminTaxonomy.ts:9]: This file is not related to Commerce, rename the file as well

```suggestion
class HeadlessAdminTaxonomy {
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by joanagrego** [comment]
ci:test:sf

**Review by kevenleone — APPROVED** [review]
Good job @joanagrego

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward:force

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by kevenleone** [comment]
ci:stop

**Comment by kevenleone** [comment]
resent: https://github.com/brianchandotcom/liferay-portal/pull/149879
